### PR TITLE
Agency support

### PR DESCRIPTION
**Pull request by gtema**

### CODE REVIEW

File: otcextensions/common/agency_auth.py [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+        agency = {'xrole_name': self.target_agency_name}
+
+        if self.target_domain_id:
+            agency['domain_id'] = self.target_domain_id
+        elif self.target_domain_name:
+            agency['domain_name'] = self.target_domain_name
+
+        return 'assume_role', agency
+
+    def get_cache_id_elements(self):
```
Comment (vladimirhasko) [inline otcextensions/common/agency_auth.py:61]: what is this method about?

File: otcextensions/common/agency_auth.py [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+        # First do regular authorization
+        auth_access = super(Agency, self).get_auth_ref(session, **kwargs)
+        # And now reauth with another scope
+        headers = {
+            'Accept': 'application/json',
+            'X-Auth-Token': auth_access._auth_token
+        }
+        body = {'auth': {'identity': {}}}
+        ident = body['auth']['identity']
+        rkwargs = {}
```
Comment (vladimirhasko) [inline otcextensions/common/agency_auth.py:93]: what is the purpose of rkwargs?

**Comment by tischrei** [comment]
I stop development in the credential due to the fact this PR is existing (didn't notice it ). So review can be finished.

File: otcextensions/common/agency_auth.py (hunk 1 above)
Comment (gtema) [reply otcextensions/common/agency_auth.py:61]: Part of the keystoneauth interface to enable "authorization caching"

File: otcextensions/common/agency_auth.py (hunk 2 above)
Comment (gtema) [reply otcextensions/common/agency_auth.py:93]: copied from existing pwd authorization. Not really used. Let's cleanup in a followup

**Review by vladimirhasko — APPROVED** [review]

**Pull request merged**